packages/core-data/src/actions.js

```diff
@@ -139,6 +139,70 @@ export function receiveEmbedPreview( url, preview ) {
 	};
 }
 
+/**
+ * Action triggered to delete an entity record.
+ *
+ * @param {string} kind      Kind of the deleted entity.
+ * @param {string} name      Name of the deleted entity.
+ * @param {Object} recordId  Record to be deleted.
+ */
+export function* deleteEntityRecord( kind, name, recordId ) {
+	const entities = yield getKindEntities( kind );
+	const entity = find( entities, { kind, name } );
+	if ( ! entity ) {
+		return;
+	}
+
+	yield {
+		type: 'DELETE_ENTITY_RECORD_START',
+		kind,
+		name,
+		recordId,
+	};
```

Comment:
Shouldn't this be used in the reducer?

Reply:
See this [comment](https://github.com/WordPress/gutenberg/pull/21557#discussion_r434035095)